**Question 3**

Let's use the model!

Write a script for loading the pipeline with pickle

Score this record:
{
    "lead_source": "paid_ads",
    "number_of_courses_viewed": 2,
    "annual_income": 79276.0
}

What's the probability that this lead will convert?

- 0.333
- **0.533**
- 0.733
- 0.933

If you're getting errors when unpickling the files, check their checksum:

$ md5sum pipeline_v1.bin
7d17d2e4dfbaf1e408e1a62e6e880d49 *pipeline_v1.bin

In [1]:
import pickle

# 1. Load the pipeline
with open('pipeline_v1.bin', 'rb') as f:
    model = pickle.load(f)

# 2. Input record
record = {
    "lead_source": "paid_ads",
    "number_of_courses_viewed": 2,
    "annual_income": 79276.0
}

# 3. Convert record to list of dicts and predict
X = [record]
y_pred = model.predict_proba(X)[0, 1]

print("Probability that this lead will convert:", y_pred)

Probability that this lead will convert: 0.5336072702798061


## Question 4

Now let's serve this model as a web service

* Install FastAPI
* Write FastAPI code for serving the model
* Now score this client using `requests`:

```python
url = "YOUR_URL"
client = {
    "lead_source": "organic_search",
    "number_of_courses_viewed": 4,
    "annual_income": 80304.0
}
requests.post(url, json=client).json()
```

What's the probability that this client will get a subscription?

* 0.334
* **0.534**
* 0.734
* 0.934
